src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,65 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE;
+
+    private T[] elementData;
+    private int size;
+
+    public MateStack() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     public void push(T value) {
+        if (value == null) {
+            return;
+        }
+
+        resize();
+        elementData[size] = value;
+        size++;
+    }
```

Comment:
why do always call resize() method? you should only resize if `size + 1 == currentMaxSize`. 
Consider adding: 
`size` variable to keep track of the current size
`currentMaxSize` variable to keep track of currently maximum size.